src/contracts/facilitators/gsm/GhoReserve.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {IERC20} from '@aave/core-v3/contracts/dependencies/openzeppelin/contracts/IERC20.sol';
+import {Ownable} from '@openzeppelin/contracts/access/Ownable.sol';
+import {VersionedInitializable} from '@aave/core-v3/contracts/protocol/libraries/aave-upgradeability/VersionedInitializable.sol';
+import {IGhoReserve} from './interfaces/IGhoReserve.sol';
+
+/**
+ * @title GhoReserve
+ * @author Aave/TokenLogic
+ * @notice It allows approved entities to withdraw and return GHO funds, with a defined maximum withdrawal capacity per entity.
+ * @dev To be covered by a proxy contract.
+ */
+contract GhoReserve is Ownable, VersionedInitializable, IGhoReserve {
+  /// @inheritdoc IGhoReserve
+  address public immutable GHO_TOKEN;
+
+  /// Map of entities and their assigned capacity and amount of GHO used
+  mapping(address => GhoUsage) private _ghoUsage;
+
+  /**
+   * @dev Constructor
+   * @param initialOwner The address of the owner
+   * @param ghoAddress The address of the GHO token on the remote chain
+   */
+  constructor(address initialOwner, address ghoAddress) Ownable() {
+    require(initialOwner != address(0), 'ZERO_ADDRESS_NOT_VALID');
+    require(ghoAddress != address(0), 'ZERO_ADDRESS_NOT_VALID');
+
+    _transferOwnership(initialOwner);
+    GHO_TOKEN = ghoAddress;
+  }
+
+  /**
+   * @dev Initializer
+   * @param newOwner The address of the new owner
+   */
+  function initialize(address newOwner) external initializer {
+    require(newOwner != address(0), 'ZERO_ADDRESS_NOT_VALID');
+    _transferOwnership(newOwner);
+  }
+
+  /// @inheritdoc IGhoReserve
+  function use(uint256 amount) external {
+    GhoUsage storage entity = _ghoUsage[msg.sender];
+    require(entity.limit >= entity.used + amount, 'LIMIT_REACHED');
```

Comment:
```suggestion
    require(entity.limit > entity.used + amount, 'LIMIT_EXCEEDED');
```
it should be > right? because if 100 is allocated, the entity can use 100.